fablabjoinville/groselha
# Create interface to register users and add React

Hi,
**Long time no see.**
Here an interface to register users is implemented with the functionality to ADD (not edit, not delete) them (yet).
Also, this view was done with React, so React is also added here, together with webpack and babel

Another thing done was the creation of a `config_db.sh` file, so that project newbies do not need to have the pain to set up the database.

For last, input validation in the form is being done, in my opinion, in a fairly manual manner. Perhaps React experts can help me out here... (@alfakini, you)
 
### Steps to test

1) Setup database
```sh
/bin/bash config_db.sh
```
then, in the python shell
```py
from app import db
db.create_all()
```

2) Install npm modules and start webpack server.
```sh
cd assets
npm install
npm  run watch
```
3) In the root dir, and in another terminal screen, type `FLASK DEBUG=true python app.py`. An error regarding env vars in `app.py` will probably appear, if so, just enter `app.py` and comment the line requiring the env var, we won't be needing it to test this feature.
4) Visit `<your-path>/register-macs` and have fun creating new users.

### Demo
![groselha](https://user-images.githubusercontent.com/21130697/38169922-2e9d2d30-354e-11e8-94d3-3aa2cfef16d6.gif)
